crates/e2e/tests/e2e/quoting.rs

```diff
@@ -0,0 +1,156 @@
+        })
+        .await
+        .unwrap();
+
+    let with_hooks = services
+        .submit_quote(&OrderQuoteRequest {
+            app_data: OrderCreationAppData::Full {
+                full: serde_json::to_string(&json!({
+                    "backend": {
+                        "hooks": {
+                            "pre": [
+                                {
+                                    "target": "0x0000000000000000000000000000000000000000",
+                                    "callData": "0x",
+                                    "gasLimit": "5000",
+                                },
+                            ],
+                            "post": [
+                                {
+                                    "target": "0x0000000000000000000000000000000000000000",
+                                    "callData": "0x",
+                                    "gasLimit": "5000",
+                                },
+                            ],
+                        },
+                    },
+                }))
+                .unwrap(),
+            },
+            ..request.clone()
+        })
+        .await
+        .unwrap();
+
+    let with_both = services
+        .submit_quote(&OrderQuoteRequest {
+            signing_scheme: QuoteSigningScheme::Eip1271 {
+                onchain_order: false,
+                verification_gas_limit: 50_000,
+            },
+            app_data: OrderCreationAppData::Full {
+                full: serde_json::to_string(&json!({
+                    "backend": {
+                        "hooks": {
+                            "pre": [
+                                {
+                                    "target": "0x0000000000000000000000000000000000000000",
+                                    "callData": "0x",
+                                    "gasLimit": "5000",
+                                },
+                            ],
+                            "post": [
+                                {
+                                    "target": "0x0000000000000000000000000000000000000000",
+                                    "callData": "0x",
+                                    "gasLimit": "5000",
+                                },
+                            ],
+                        },
+                    },
+                }))
+                .unwrap(),
+            },
+            ..request.clone()
+        })
+        .await
+        .unwrap();
+
+    let base = services.submit_quote(&request).await.unwrap();
+
+    tracing::debug!(
+        ?with_eip1271,
+        ?with_hooks,
+        ?with_both,
+        ?base,
+        "Computed quotes."
+    );
+
+    assert!(base.quote.fee_amount < with_eip1271.quote.fee_amount);
+    assert!(base.quote.fee_amount < with_hooks.quote.fee_amount);
+
+    assert!(with_both.quote.fee_amount > with_hooks.quote.fee_amount);
+    assert!(with_both.quote.fee_amount > with_hooks.quote.fee_amount);
```

Comment:
Copy and paste error.

Reply:
Dang... https://github.com/cowprotocol/services/pull/1650